src/core/graph.js

```diff
@@ -1212,3 +1212,73 @@ export function edgeToParts(
   const timestampMs = edge.timestampMs;
   return {addressParts, srcParts, dstParts, timestampMs};
 }
+
+export function compareGraphs(
+  firstGraph: Graph,
+  secondGraph: Graph
+): {|
+  +isEqual: boolean,
+  +uniqueNodesInFirst: Node[],
+  +uniqueNodesInSecond: Node[],
+  +nodeTuplesWithDifferences: [Node, Node][],
+  +uniqueEdgesInFirst: Edge[],
+  +uniqueEdgesInSecond: Edge[],
+  +edgeTuplesWithDifferences: [Edge, Edge][],
```

Comment:
Recommend changing all the arrays to read-only arrays, e.g. `s/Node[]/$ReadOnlyArray<Node>/`.

there's no reason why a caller would want the ability to mutate the graph comparison result, since doing so would just make the comparison inaccurate.

Reply:
done